tests/ssl-opt.sh

```diff
@@ -1013,6 +1031,46 @@ cleanup() {
 
 get_options "$@"
 
+# Optimize filters: if $FILTER and $EXCLUDE can be expressed as shell
+# patterns rather than regular expressions, use a case statement instead
+# of calling grep. To keep the optimizer simple, it is incomplete and only
+# detects simple cases: plain substring, everything, nothing.
+#
+# As an exception, the character '.' is treated as an ordinary character
+# if it is the only special character in the string. This is because it's
+# rare to need "any one character", but needing a literal '.' is common
+# (e.g. '-f "DTLS 1.2"').
+need_grep=
+case "$FILTER" in
```

Comment:
I tested it and only the `[` and `]` are working from the BRE. I think here is the problem because the `*` working as a wildcard and the `?` matches any single character as expected from the matching in the `case` instruction.

Reply:
Sorry, I don't understand. Could you give a counterexample?

Reply:
Tried with these which should be working and resulted with filtered out everything:
`.*ECDSA`, `.*ECDSA.*`, `AES.*SHA`, `^PSA`, `(G|C)CM`, `(srv|cli)`, `(srv)`, `C{2,3}`

Tried with these which should not and does have some filtered result:
`AES?256`, `AES?*256`

This is worked as expected:
`[CG]CM`

Reply:
Fixed, thanks for noticing.